fission-web-server/library/Fission/Web/Server/Types.hs

```diff
@@ -140,6 +155,102 @@ instance MonadDB (Transaction Server) Server where
+
+  set recType url zoneTxt contents ttl = do
+    logDebug $ mconcat
+      [ "Set DNS "
+      , displayShow recType
+      , " Record "
+      , displayShow url
+      , " to "
+      , displayShow contents
+      ]
+    clientEnv <- getClientEnv
+    let
+      rrset = RRSets [RRSet { rrset_name = Text.append (textDisplay url) "."
+                      , rrset_type = recType
+                      , rrset_ttl = ttl
+                      , rrset_changetype = Just Replace
+                      , rrset_records = Just $ formatRecords contents
+                      , rrset_comments = Nothing
+      }]
+    resp <- liftIO $ runClientM (PDNS.updateRecords "localhost" zoneTxt rrset) clientEnv
+    case resp of
+      Left err -> do
+        logWarn @Text "PowerDNS.set failed"
+        return . Left $ Web.Error.toServerError err
+      Right good -> do
+        logDebug @Text "PowerDNS.set succeeded"
+        return $ Right good
+    where
+      formatRecords :: NonEmpty Text -> [Record]
+      formatRecords values =
+        fmap newRecord (toList values)
+
+      newRecord :: Text -> Record
+      newRecord content =
+        Record { record_content = format recType content, record_disabled = False }
+
+      format :: PowerDNS.API.Zones.RecordType -> Text -> Text
+      format TXT = UTF8.wrapIn "\""
+      format _   = identity
+
+  get url _ = do
+    logDebug $ "PowerDNS.get for " <> textDisplay url
+    clientEnv <- getClientEnv
+    resp <- liftIO $ runClientM (PDNS.search "localhost" (textDisplay url) 1 (Just TyRecord)) clientEnv
+    case resp of
+      Left err -> do
+        logWarn @Text "PowerDNS.get failed"
+        return . Left $ Web.Error.toServerError err
+      Right good -> do
+        logDebug @Text "PowerDNS.get succeeded"
+        return $ Right good
+
+  clear url zoneID = do
+    logDebug $ "PowerDNS.clear " <> textDisplay url
+    results <- PowerDNS.get url zoneID
+    case results of
+      Left err -> do
+        return $ Left err
+      Right searchResults -> do
+        mapM_ removeResult searchResults
+        return $ Right ()
+    where
+      removeResult :: (MonadIO m, MonadPowerDNS m) => PDNS.SearchResult -> m ()
+      removeResult searchResult = do
+        clientEnv <- getClientEnv
+        let
+          recType = convertType (PDNS.sr_type searchResult)
+          rrset = RRSets [RRSet { rrset_name = PDNS.sr_name searchResult
+                      , rrset_type = recType
+                      , rrset_ttl = 10
+                      , rrset_changetype = Just PowerDNS.API.Zones.Delete
+                      , rrset_records = Nothing
+                      , rrset_comments = Nothing
+          }]
+        _ <- liftIO $ runClientM (PDNS.updateRecords "localhost" zoneID rrset) clientEnv
+        return ()
+
+      convertType :: Text -> PowerDNS.API.Zones.RecordType
+      convertType "CNAME" = CNAME
+      convertType "TXT"   = TXT
+      convertType _       = PowerDNS.API.Zones.A
```

Comment:
This is totally fine, but some background:

This is typically covered by the `Read` typeclass, but it looks like they didn't implement that. You can plug into their parser by leveraging the JSON instances:

```haskell
decode (toJSON txt) :: Maybe RecordType

-- With a fallback to an A record, though not totally sure that's what you want?

maybe Zones.A $ decode $ toJSON txt :: RecordType
```